### ISSUE TITLE
About trianing

### ISSUE THREAD

**Issue by YinWeiling** [liudaizong/F2Net#2]
Thanks for your great work!!!
I have a question about the training strategy. The paper says "During training the Center Guiding Appearance Diffusion
module, we first separately train the Center Prediction Branch and Spatial-Prior Guided Appearance Matching in parallel in the first 20 epochs, where we feed the appearancematching module with the center point of ground truth." But I could not find the code for training the Center Prediction Branch and Spatial-Prior Guided Appearance Matching. Does ie means that for the first 20 epochs, the code below is used to train the network? Or the code is somewhere else but I didn't find. Could you please locate it ?
`if i_iter%3 ==0: #对于静态图片的训练
              
                pred1, temp, heat = model(images, images, images, img_ct, img_guass)
                heat = _sigmoid(heat)
                loss = 0.1*(loss_calc1(temp, labels) + 0.8* loss_calc2(temp, labels))+ loss_calc3(heat, img_guass)
                loss.backward()
`

**Comment by liudaizong** [comment]
> Thanks for your great work!!! I have a question about the training strategy. The paper says "During training the Center Guiding Appearance Diffusion module, we first separately train the Center Prediction Branch and Spatial-Prior Guided Appearance Matching in parallel in the first 20 epochs, where we feed the appearancematching module with the center point of ground truth." But I could not find the code for training the Center Prediction Branch and Spatial-Prior Guided Appearance Matching. Does ie means that for the first 20 epochs, the code below is used to train the network? Or the code is somewhere else but I didn't find. Could you please locate it ? `if i_iter%3 ==0: #对于静态图片的训练
> 
> ```
>             pred1, temp, heat = model(images, images, images, img_ct, img_guass)
>             heat = _sigmoid(heat)
>             loss = 0.1*(loss_calc1(temp, labels) + 0.8* loss_calc2(temp, labels))+ loss_calc3(heat, img_guass)
>             loss.backward()
> ```
> 
> `

Thanks for your interest in our work. There is no code for two-step training. You can train 20 epochs in the first step  and save the model, then load the model to train the second step. We also utilize such manual adjustment to train our model.

**Comment by YinWeiling** [comment]
> > Thanks for your great work!!! I have a question about the training strategy. The paper says "During training the Center Guiding Appearance Diffusion module, we first separately train the Center Prediction Branch and Spatial-Prior Guided Appearance Matching in parallel in the first 20 epochs, where we feed the appearancematching module with the center point of ground truth." But I could not find the code for training the Center Prediction Branch and Spatial-Prior Guided Appearance Matching. Does ie means that for the first 20 epochs, the code below is used to train the network? Or the code is somewhere else but I didn't find. Could you please locate it ? `if i_iter%3 ==0: #对于静态图片的训练
> > ```
> >             pred1, temp, heat = model(images, images, images, img_ct, img_guass)
> >             heat = _sigmoid(heat)
> >             loss = 0.1*(loss_calc1(temp, labels) + 0.8* loss_calc2(temp, labels))+ loss_calc3(heat, img_guass)
> >             loss.backward()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > `
> 
> Thanks for your interest in our work. There is no code for two-step training. You can train 20 epochs in the first step and save the model, then load the model to train the second step. We also utilize such manual adjustment to train our model.

Thank you, I will manually adjust the code.

**Comment by liudaizong** [comment]
> > > Thanks for your great work!!! I have a question about the training strategy. The paper says "During training the Center Guiding Appearance Diffusion module, we first separately train the Center Prediction Branch and Spatial-Prior Guided Appearance Matching in parallel in the first 20 epochs, where we feed the appearancematching module with the center point of ground truth." But I could not find the code for training the Center Prediction Branch and Spatial-Prior Guided Appearance Matching. Does ie means that for the first 20 epochs, the code below is used to train the network? Or the code is somewhere else but I didn't find. Could you please locate it ? `if i_iter%3 ==0: #对于静态图片的训练
> > > ```
> > >             pred1, temp, heat = model(images, images, images, img_ct, img_guass)
> > >             heat = _sigmoid(heat)
> > >             loss = 0.1*(loss_calc1(temp, labels) + 0.8* loss_calc2(temp, labels))+ loss_calc3(heat, img_guass)
> > >             loss.backward()
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > `
> > 
> > 
> > Thanks for your interest in our work. There is no code for two-step training. You can train 20 epochs in the first step and save the model, then load the model to train the second step. We also utilize such manual adjustment to train our model.
> 
> Thank you, I will manually adjust the code.

If things go right, I'll close this issue. Feel free to reopen it if there is still problem.
